demo/midgard-sdk/src/settlement.ts

```diff
@@ -0,0 +1,513 @@
+    }),
+  }),
+  Data.Object({
+    Remove: Data.Object({
+      settlementId: Data.Bytes(),
+      inputIndex: Data.Integer(),
+      spendRedeemerIndex: Data.Integer(),
+    }),
+  }),
+]);
+export type SettlementMintRedeemer = Data.Static<
+  typeof SettlementMintRedeemerSchema
+>;
+export const SettlementMintRedeemer =
+  SettlementMintRedeemerSchema as unknown as SettlementMintRedeemer;
+
+export const ActiveOperatorSpendRedeemerSchema = Data.Enum([
+  Data.Literal("ListStateTransition"),
+  Data.Object({
+    UpdateBondHoldNewState: Data.Object({
+      activeNodeOutputIndex: Data.Integer(),
+      hubOracleRefInputIndex: Data.Integer(),
+      stateQueueRedeemerIndex: Data.Integer(),
+    }),
+  }),
+  Data.Object({
+    UpdateBondHoldNewSettlement: Data.Object({
+      activeNodeOutputIndex: Data.Integer(),
+      hubOracleRefInputIndex: Data.Integer(),
+      settlementQueueInputIndex: Data.Integer(),
+      settlementQueueRedeemerIndex: Data.Integer(),
+      newBondUnlockTime: POSIXTimeSchema,
+    }),
+  }),
+]);
+export type ActiveOperatorSpendRedeemer = Data.Static<
+  typeof ActiveOperatorSpendRedeemerSchema
+>;
+export const ActiveOperatorSpendRedeemer =
+  ActiveOperatorSpendRedeemerSchema as unknown as ActiveOperatorSpendRedeemer;
+
+export const ActiveOperatorSpendDatumSchema = Data.Object({
+  key: Data.Nullable(Data.Bytes()),
+  link: Data.Nullable(Data.Bytes()),
+  bondUnlockTime: Data.Nullable(POSIXTimeSchema),
+});
+export type ActiveOperatorSpendDatum = Data.Static<
+  typeof ActiveOperatorSpendDatumSchema
+>;
+export const ActiveOperatorSpendDatum =
+  ActiveOperatorSpendDatumSchema as unknown as ActiveOperatorSpendDatum;
+
+export type AttachResolutionClaimParams = {
+  settlementAddress: string;
+  resolutionClaimOperator: string;
+  newBondUnlockTime: bigint;
+  hubOracleValidator: AuthenticatedValidator;
+  schedulerScriptAddress: string;
+  schedulerPolicyId: string;
+};
+
+export type SettlementUTxO = {
+  utxo: UTxO;
+  datum: SettlementDatum;
+};
+
+export const getSettlementUTxOWithoutClaim = (
+  settlementUTxOs: UTxO[],
+): Effect.Effect<SettlementUTxO, DataCoercionError | LucidError> => {
+  for (const utxo of settlementUTxOs) {
+    const datumCBOR = utxo.datum;
+    if (!datumCBOR) {
+      continue;
+    }
+    try {
+      const parsedDatum = Data.from(datumCBOR, SettlementDatum);
+      if (parsedDatum.resolutionClaim === null) {
+        return Effect.succeed({ utxo, datum: parsedDatum });
+      }
+    } catch (err) {
+      return Effect.fail(
+        new DataCoercionError({
+          message: `Could not coerce UTxO's datum to a settlement datum`,
+          cause: err,
+        }),
+      );
+    }
+  }
+  return Effect.fail(
+    new LucidError({
+      message: "No settlement UTxO without resolution claim found",
+      cause: "Settlement Tx not initiated",
+    }),
+  );
+};
+// change to a single ref utxo instead of list
+const fetchHubOracleRefUTxOs = (
```

Comment:
Nice, we can rename this to its singular form now :)
```suggestion
const fetchHubOracleRefUTxO = (
```